velox/exec/Spill.cpp

```diff
@@ -23,10 +23,9 @@ namespace facebook::velox::exec {
 namespace {
 // Spilling currently uses the default PrestoSerializer which by default
 // serializes timestamp with millisecond precision to maintain compatibility
-// with presto. Since velox's native timestamp implementation supports
-// nanosecond precision, we use this serde option to ensure the serializer
-// preserves precision.
-static const bool kDefaultUseLosslessTimestamp = true;
+// with presto.
+// There is a bug in LosslessTimestamp serialization see issue #6837.
```

Comment:
Instead of turning it off, can you just remove this logic? We can implement proper solution later.

CC: @rui-mo

Reply:
Also, should we add a check that we are not loosing any data during serialization and fail if we do?

Reply:
(1) I can remove the flag, but the logic is propagated across the function I do not think we should remove it if we plan to add it.

for (2) I can add such check, but this could actually expose issues whenever a function or anywhere int the  code we do generate a timestamp that is not of  milli precision. are you sure we shall do that? do we have functions that might confirm to that? I Wonder if we should different time stamp types for presto and non presto,
why do we  need to store nanos if we want to enforce in memory milli precision for presto.

Reply:
its loo late to catch those at spilling time instead of construction time